### PR TITLE
feat: adds find all selectors to test utils

### PR DESCRIPTION
**Pull request by orangevolon**
### Description

Adds two `findAll[COMPONENT_NAME]s()` test utils selectors to each component wrapper.

Depends on https://github.com/cloudscape-design/test-utils/pull/74

Related links, issue #, if available: **Test utils API improvements project**

### How has this been tested?

✅  Copies including the JSDocs and the pluralizations have been reviewed by the content team.
🧪 Unit tests added to both selectors and dom test utils.

Related links, issue #, if available: Test utils API improvements project

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -14765,8 +14765,9 @@
     },
     "node_modules/pluralize": {
       "version": "8.0.0",
+      "resolved": "https://registry.npmjs.org/pluralize/-/pluralize-8.0.0.tgz",
+      "integrity": "sha512-Nc3IT5yHzflTfbjgqWcCPpo7DaKy4FnpB0l/zCAW0Tc7jxAiuqSxHasntB3D7887LSrA93kDJ9IXovxJYxyLCA==",
```
Comment (johannes-weber) [inline package-lock.json:14769]: This looks like a leftover as we're not using the pluralize package.

File: src/__tests__/functional-tests/test-utils.test.tsx [hunk 2]
```diff
@@ -7,9 +7,76 @@
...
+  // For this reason, looking into DOM is representative of both groups.
+  const wrapperPropsList = Object.keys(DomElementWrapper.prototype);
+
+  // Every component has the same set of selector functions.
+  // For this reason, casting the function names into the Alert component.
+  const findName = `find${componentNamePascalCase}` as 'findAlert';
+  const findAllName = wrapperPropsList.find(selector => findAllRegex.test(selector)) as 'findAllAlerts';
+  const findByTestIdName = `find${componentNamePascalCase}ByTestId` as 'findAlertByTestId';
+
+  return { findName, findAllName, findByTestIdName };
```
Comment (johannes-weber) [inline src/__tests__/functional-tests/test-utils.test.tsx:78]: `findByTestIdName` is not used anymore and should be removed.

**Review by johannes-weber — CHANGES_REQUESTED** [review]

File: package-lock.json (hunk 1 above)
Comment (orangevolon) [reply package-lock.json]: We don't have any package json change. Discarding this file altogether.

File: src/__tests__/functional-tests/test-utils.test.tsx [hunk 3]
```diff
@@ -187,40 +187,46 @@ describe.each(components)('ElementWrapper selectors for %s component', component
   });
 });
 
-describe.each(componentsWithNoExtraFinders)('ElementWrapper selectors for %s component', componentName => {
+describe.only.each(componentswithseveralroots)('ElementWrapper selectors for %s component', componentName => {
```
Comment (orangevolon) [inline src/__tests__/functional-tests/test-utils.test.tsx:190]: These two components `<AppLayout />` and `<TopNavigation />` will still have some limited functionality for the same reason [mentioned here](https://github.com/cloudscape-design/components/pull/2932/files#r1831055671).

These two cases will not work:

1. Adding custom selector in unit tests:
```tsx
findAllAppLayouts('[data-my-custom-attrib]')

// Generated selector:
// .awsui_root_hash[data-my-custom-attrib]

<div data-my-custom-attrib>   // <--- data attribute is added here
  <div class="awsui_root_hash"> // <--- actual root class is here
    ...
  </div>
</div>
```

2. Using `get()` in integration tests:
```tsx
findAllAppLayouts().get(2)

// Generated selector
// .awsui_root_hash:nth-child(2);

<div>
  <div class="awsui_root_hash">...</div>
</div>
<div>
  <div class="awsui_root_hash">...</div>
</div>
```

**Review by johannes-weber — APPROVED** [review]

**Pull request merged**